Sources/SocksCore/TCPSocket.swift

```diff
@@ -39,22 +39,29 @@ public protocol TCPReadableSocket: TCPSocket { }
 extension TCPReadableSocket {
 
     public func recv(maxBytes: Int = BufferCapacity) throws -> [UInt8] {
+    	var out = Array<UInt8>()
         let data = Bytes(capacity: maxBytes)
         let flags: Int32 = 0 //FIXME: allow setting flags with a Swift enum
         let receivedBytes = socket_recv(self.descriptor, data.rawBytes, data.capacity, flags)
-        guard receivedBytes > -1 else { throw SocksError(.readFailed) }
-        
-        guard receivedBytes > 0 else {
+        guard receivedBytes > -1 else {
+	    if errno == ECONNRESET { // closed by peer, need to close this side too!
```

Comment:
So, by receiving this, specific error, we're using it to indicate that it's not an error, but a proper close from opponent and we should close, but not throw an error?

Can we add a little more of a comment about that.